src/main/scala/mesosphere/marathon/core/appinfo/impl/AppInfoBaseData.scala

```diff
@@ -0,0 +1,123 @@
+package mesosphere.marathon.core.appinfo.impl
+
+import mesosphere.marathon.MarathonSchedulerService
+import mesosphere.marathon.Protos.MarathonTask
+import mesosphere.marathon.core.appinfo.{ TaskCounts, AppInfo, EnrichedTask }
+import mesosphere.marathon.health.{ Health, HealthCheckManager, HealthCounts }
+import mesosphere.marathon.state.{ TaskFailure, TaskFailureRepository, Identifiable, PathId, AppDefinition }
+import mesosphere.marathon.tasks.TaskTracker
+import mesosphere.marathon.upgrade.DeploymentPlan
+import org.slf4j.LoggerFactory
+
+import scala.collection.immutable.Seq
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+class AppInfoBaseData(
+    taskTracker: TaskTracker,
+    healthCheckManager: HealthCheckManager,
+    marathonSchedulerService: MarathonSchedulerService,
+    taskFailureRepository: TaskFailureRepository) {
+  import AppInfoBaseData.log
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  lazy val runningDeploymentsByAppFuture: Future[Map[PathId, Seq[Identifiable]]] = {
+    log.debug("Retrieving running deployments")
+
+    val allRunningDeploymentsFuture: Future[Seq[DeploymentPlan]] =
+      marathonSchedulerService.listRunningDeployments().map(_.map(_._1))
```

Comment:
something like `.map(_.map(_._1))` is hard to grasp. Perhaps listRunningDeployments can give a better result?